src/main/java/frc/robot/Robot.java

```diff
@@ -29,24 +16,16 @@
  * creating this project, you must also update the build.gradle file in the
  * project.
  */
-@SuppressWarnings("all")
 public class Robot extends TimedRobot {
+
   private Command m_autonomousCommand;
```

Comment:
Why not make this an AutoCommandInterface? Or at least declare an AutoCommandInterface in addition to this so we don'thave to declare the variable locally in disablePeriodic?

Reply:
I strongly believe that you should not use an instance variable for something in a method, unless there is a real reason. In disabledPeriodic, there is no reason to save that value, so  it might as well be a local variable. 
In autonomousInit, it needs to be saved so that it can be canceled in teleopInit, so it needs to be an instance variable.

Reply:
Also, we should not be setting it into m_autonomousCommand in disabledPeriodic, because that implies that is is running. It is only turned on in autoInit.